libs/domain/cart/cart-remove/cart-remove.component.ts

```diff
@@ -0,0 +1,92 @@
+import { CartService } from '@spryker-oryx/cart';
+import { resolve } from '@spryker-oryx/di';
+import { NotificationService } from '@spryker-oryx/site';
+import { AlertType } from '@spryker-oryx/ui';
+import { ButtonColor, ButtonSize, ButtonType } from '@spryker-oryx/ui/button';
+import { IconTypes } from '@spryker-oryx/ui/icon';
+import { I18nMixin, hydrate } from '@spryker-oryx/utilities';
+import { LitElement, TemplateResult, html } from 'lit';
+import { state } from 'lit/decorators.js';
+import { CartComponentMixin } from '../src/mixins';
+
+@hydrate({ event: ['mouseover', 'focusin'] })
+export class CartRemoveComponent extends CartComponentMixin(
+  I18nMixin(LitElement)
+) {
+  protected cartService = resolve(CartService);
+  protected notificationService = resolve(NotificationService);
+
+  @state() protected requestsConfirmation = false;
+  @state() protected isLoading = false;
+
+  protected override render(): TemplateResult | void {
+    if (!this.$cart()?.id) return;
+
+    return html`
+      <oryx-button
+        .type=${ButtonType.Icon}
+        .size=${ButtonSize.Md}
+        .label=${this.i18n('carts.remove')}
+        .icon=${IconTypes.Trash}
+        @click=${this.onRemove}
+      ></oryx-button>
+      ${this.renderRemoveConfirmationModal()}
+    `;
+  }
+
+  protected onRemove(): void {
+    this.requestsConfirmation = true;
+  }
+
+  protected renderRemoveConfirmationModal(): TemplateResult | void {
+    if (!this.requestsConfirmation) return;
+
+    const name = this.$cart()?.name;
+
+    return html`<oryx-modal
+      ?open=${this.requestsConfirmation}
+      .heading=${this.i18n('carts.remove.title.confirmation')}
+      @oryx.modal.closed=${this.onCancel}
+      enableFooter
+    >
+      <span>${this.i18n('cart.remove.remove-cart-<name>-info', { name })}</span>
+      <oryx-button
+        slot="footer-more"
+        .color=${ButtonColor.Error}
+        .size=${ButtonSize.Md}
+        .text=${this.i18n('carts.remove.button.remove')}
+        ?loading=${this.isLoading}
+        @click=${this.onConfirm}
+      ></oryx-button>
+    </oryx-modal>`;
+  }
+
+  protected onConfirm(): void {
+    this.isLoading = true;
```

Comment:
```suggestion
    this.isRemoving = true;
```

Seems more accurate. It could also be used to start removing the row (animation).

Reply:
then `inProgress` will be even more clear